File: eco-feature/eco-feature-workspace/src/main/react/graphql/workspace.ts
export interface Workspace {
  id: string
  name: string
  variables: KeyValue[]
  users: WorkspaceUser[]
  host?: string
}

export interface WorkspaceUser {
  id: string
  name: string
  role: string
}

export interface WorkspaceRole {
  name: string
}

export interface WorkspaceInput {
  name: string
  image?: WorkspaceImageInput
  variables: KeyValueInput[]
  host?: string
}

export interface WorkspaceImageInput {
  file: string
  name: string
}

export interface WorkspaceUserInput {
  reference: string
  role: string
}

export interface KeyValue {
  key: string
  value?: string
}

export interface KeyValueInput {
  key: string
  value?: string
}

File: eco-feature/eco-feature-workspace/src/main/kotlin/model/Workspace.kt
package community.flock.eco.feature.workspace.model

import community.flock.eco.core.model.AbstractEntity
import java.util.*
import javax.persistence.ElementCollection
import javax.persistence.Embedded
import javax.persistence.Entity
import javax.persistence.Id

@Entity
data class Workspace(

    @Id
    override val id: UUID = UUID.randomUUID(),

    val name: String,

    @Embedded
    val image: WorkspaceImage? = null,

    @ElementCollection
    val variables: Map<String, String?> = mapOf(),

    val host: String? = null,

    @ElementCollection
    val users: Set<WorkspaceUserRole> = setOf()

) : AbstractEntity<UUID>(id)
